Adding pluginv2 support for libnetwork.

anusha-ragunathan opened the pull request:
Legacy plugins (aka pluginv1) calls in libnetwork are replaced with
calls using the new plugin model (aka pluginv2). pkg/plugins is still
used for managing the http client connections to the plugin.

This commit makes the necessary changes in docker/docker. Part 2 will
will take care of the libnetwork changes.

Signed-off-by: Anusha Ragunathan anusha@docker.com

mavenugo commented on vendor/src/github.com/docker/libnetwork/controller.go:
```diff
@@ -1003,9 +1003,9 @@ func SandboxKeyWalker(out *Sandbox, key string) SandboxWalker {
 	// Plugins pkg performs lazy loading of plugins that acts as remote drivers.
 	// As per the design, this Get call will result in remote driver discovery if there is a corresponding plugin available.
-	_, err := plugins.Get(networkType, driverapi.NetworkPluginEndpointType)
+	_, err := pluginstore.LookupWithCapability(networkType, driverapi.NetworkPluginEndpointType)
```
Before doing a deep dive into the pluginstore APIs, will this call work in backward compatible way by loading lazy plugins (based of plugin-v1) ?

anusha-ragunathan replied:
Yes. The approach is to try with pluginv2 first. If that fails, fallback to pluginv1. The fallback is controlled by allowV1PluginsFallback, which will be true for 1.13.

anusha-ragunathan committed:
Adding pluginv2 support for libnetwork (part 1)

Legacy plugins (aka pluginv1) calls in libnetwork are replaced with
calls using the new plugin model (aka pluginv2). pkg/plugins is still
used for managing the http client connections to the plugin.

This commit makes the necessary changes in docker/docker. Part 2 will
will take care of the libnetwork changes.

Signed-off-by: Anusha Ragunathan <anusha@docker.com>

vieux commented:
LGTM

cpuguy83 commented:
LGTM
